File: new/dart-maor/done-today-for_test.dart
// בדיקת-חוזה (רתמת-זהב) · doneTodayFor — מייבאת אך ורק את האטום-שלה (חוק-4).
// שש דוגמאות-החוזה זהות ביט-אחר-ביט למקור-ה-JS new/atoms/done-today-for.test.mjs,
// עם אותו שקע-זהות מוזרק (trim+lowercase; ריק/חסר ⇒ 'מקומי'):
//   1) [{A@x.com, 2026-08-24T10:00}], 'a@x.com' ⇒ 1  (רישיות מנורמלת)
//   2) [{a@x.com, 2026-08-23T23:59}], 'a@x.com' ⇒ 0  (אתמול)
//   3) [{a@x.com}],                    'a@x.com' ⇒ 0  (פתוחה — אין doneAt)
//   4) [{b@x.com, 2026-08-24}],        'a@x.com' ⇒ 0  (עובד/ת אחר/ת)
//   5) [{doneAt:2026-08-24},{assignee:'', doneAt:2026-08-24}], '' ⇒ 2  (ריק⇒'מקומי')
//   6) תערובת 1–4 יחד,                'a@x.com' ⇒ 1
// הרצה: dart run --enable-asserts new/dart-maor/done-today-for_test.dart  ⇒ exit 0
import 'done-today-for.dart';

void _ok(bool cond, String msg) {
  if (!cond) throw StateError('FAIL: $msg');
}

// שקע-הזהות החוזי (task-identity), מוזרק בדיוק כמו בבדיקת-ה-JS.
String _taskIdentity(Object? email) {
  final e = ((email ?? '') as String).trim().toLowerCase();
  return e.isEmpty ? 'מקומי' : e;
}

const String _today = '2026-08-24';

void main() {
  var n = 0;

  final t1 = <String, Object?>{'assignee': 'A@x.com', 'doneAt': '2026-08-24T10:00'};
  final t2 = <String, Object?>{'assignee': 'a@x.com', 'doneAt': '2026-08-23T23:59'};
  final t3 = <String, Object?>{'assignee': 'a@x.com'};
  final t4 = <String, Object?>{'assignee': 'b@x.com', 'doneAt': '2026-08-24'};

  // כל שורה: [tasks, identity, expected]
  final cases = <List<Object?>>[
    [
      [t1], 'a@x.com', 1
    ],
    [
      [t2], 'a@x.com', 0
    ],
    [
      [t3], 'a@x.com', 0
    ],
    [
      [t4], 'a@x.com', 0
    ],
    [
      [
        <String, Object?>{'doneAt': '2026-08-24'},
        <String, Object?>{'assignee': '', 'doneAt': '2026-08-24'},
      ],
      '',
      2
    ],
    [
      [t1, t2, t3, t4], 'a@x.com', 1
    ],
  ];

  for (final c in cases) {
    final tasks = (c[0] as List).cast<Map<String, Object?>>();
    final id = c[1] as String;
    final want = c[2] as int;
    final got = doneTodayFor(tasks, id, _today, _taskIdentity);
    _ok(got == want, "doneTodayFor(.., '$id') = $got ≠ $want");
    n++;
  }

  // assert חי (חוק: --enable-asserts) — מוכיח שהמנגנון פעיל.
  assert(doneTodayFor([t1], 'a@x.com', _today, _taskIdentity) == 1, 'assert-live guard');

  print('OK doneTodayFor: $n asserts passed');
}

File: new/atoms/done-today-for.test.mjs
import { doneTodayFor as __pure_doneTodayFor } from './done-today-for.mjs';
// צילום-מקומי + עטיפת-כריכה (מנוע-הקשיחים; בדיקה לא מייבאת אטום-שכן)
const __d_done_today_for_T = {
  k1: 10,
};
const doneTodayFor = (...a) => __pure_doneTodayFor(...a, ...Array(Math.max(0, 4 - a.length)).fill(undefined), __d_done_today_for_T);
// שקע מוזרק לפי חוזה task-identity (הבדיקה לא מייבאת אטום אחר — חוק-4)
const taskIdentity = (email) => { const e = (email ?? '').trim().toLowerCase(); return e || 'מקומי'; };
const TODAY = '2026-08-24';
const T1 = { assignee: 'A@x.com', doneAt: '2026-08-24T10:00' };
const T2 = { assignee: 'a@x.com', doneAt: '2026-08-23T23:59' };
const T3 = { assignee: 'a@x.com' };
const T4 = { assignee: 'b@x.com', doneAt: '2026-08-24' };
const C = [
  [[T1], 'a@x.com', 1],
  [[T2], 'a@x.com', 0],
  [[T3], 'a@x.com', 0],
  [[T4], 'a@x.com', 0],
  [[{ doneAt: '2026-08-24' }, { assignee: '', doneAt: '2026-08-24' }], '', 2],
  [[T1, T2, T3, T4], 'a@x.com', 1],
];
let f = 0;
for (const [tasks, id, w] of C) {
  const g = doneTodayFor(tasks, id, TODAY, taskIdentity);
  if (g !== w) { console.error(`✗ doneTodayFor(${JSON.stringify(tasks)}, '${id}') = ${g} ≠ ${w}`); f = 1; }
}
if (f) process.exit(1);
console.log('✓ done-today-for: 6 דוגמאות-חוזה (זהות-מוזרקת) — ירוק');
